runtime/lua/vim/fs.lua

```diff
@@ -350,6 +350,9 @@ function M.normalize(path, opts)
 
   path = path:gsub('\\', '/'):gsub('/+', '/')
 
+  if path == '/' then
+    return path
+  end
   return path:sub(-1) == '/' and path:sub(1, -2) or path
```

Comment:
I actually suspect the correct fix to this issue is in this line. This line unconditionally removes the trailing `/` on a path. Even if it's the only character.

Reply:
Would it be better if normalize always returns a trailing `/` for a directory? I.e. just `return path` ?

Reply:
That is what it did originally. It was changed to remove the trailing slash in 25fa051fa157ca4bce712e61602447a4222581ca. There are probably arguments supporting either case, honestly.

I'd be in favor of _not_ removing the trailing slash, personally.

Reply:
Thanks for reviewing!

> I actually suspect the correct fix to this issue is in this line. This line unconditionally removes the trailing / on a path. Even if it's the only character. We should fix it here instead of adding another special case.

Does this mean the following?

```diff
diff --git a/runtime/lua/vim/fs.lua b/runtime/lua/vim/fs.lua
index 28d090543..668cf80ab 100644
--- a/runtime/lua/vim/fs.lua
+++ b/runtime/lua/vim/fs.lua
@@ -350,10 +350,7 @@ function M.normalize(path, opts)
 
   path = path:gsub('\\', '/'):gsub('/+', '/')
 
-  if path == '/' then
-    return path
-  end
-  return path:sub(-1) == '/' and path:sub(1, -2) or path
+  return path:sub(-1) == '/' and path ~= '/' and path:sub(1, -2) or path
 end
 
 return M
```

Reply:
I'm suggesting simply

```lua
return path
```

Reply:
Thanks.

Some users may depend the undocumented behavior.
I added it to news.txt just in case.